### PR TITLE
EID-1289: Add configuration to the translator for the CloudHSM

### PR DESCRIPTION
**Pull request by vixus0**
- Add polymorphic configuration for `SignerConfiguration` which will
allow you to use a key file or the CloudHSM to create a
`SamlObjectSigner`
- Refactor `SamlObjectSigner` to use a `BasicX509Credential` which can
correspond to credentials from either files on disk or the CloudHSM
- `EspAuthnRequestAppRuleTest` is still failing for now (needs
investigation)

### CODE REVIEW

File: proxy-node-translator/src/dist/config.yml [hunk 1]
```diff
@@ -16,11 +16,9 @@ proxyNodeMetadataForConnectorNodeUrl: ${PROXY_NODE_METADATA_FOR_CONNECTOR_NODE_U
 connectorNodeIssuerId: ${CONNECTOR_NODE_ISSUER_ID}
 
 keyRetrieverServiceName: ${KEY_RETRIEVER_SERVICE_NAME:-softHSM}
-softHSMLibPath: ${SOFT_HSM_LIB_PATH}
-softHSMSigningKeyPin: ${SOFT_HSM_SIGNING_KEY_PIN}
-softHSMSigningKeyLabel: ${SOFT_HSM_SIGNING_KEY_LABEL}
 
-connectorFacingSigningKeyPair:
+signerConfiguration:
+  type: file
```
Comment (JHjava) [inline proxy-node-translator/src/dist/config.yml]: Should this be set as an environment variable which defaults to file or does this get overridden else where?

File: eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/base/EidasSamlParserAppRuleTestBase.java [hunk 2]
```diff
@@ -9,11 +9,13 @@
 import org.opensaml.core.xml.io.MarshallingException;
 import org.opensaml.saml.saml2.core.AuthnRequest;
 import org.opensaml.security.credential.Credential;
+import org.opensaml.security.x509.BasicX509Credential;
```
Comment (JHjava) [inline eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/base/EidasSamlParserAppRuleTestBase.java]: Doesn't look any of these new imports are being used

File: proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/configuration/TranslatorConfiguration.java [hunk 3]
```diff
@@ -69,11 +48,7 @@ public String getConnectorNodeIssuerId() {
         return connectorNodeIssuerId;
     }
 
-    public String getKeyRetrieverServiceName() { return keyRetrieverServiceName; }
-
-    public String getSoftHSMLibPath() { return softHSMLibPath; }
-
-    public String getSoftHSMSigningKeyPin() { return softHSMSigningKeyPin; }
```
Comment (JHjava) [inline proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/configuration/TranslatorConfiguration.java]: Can the rest of the Soft HSM be removed now? I.E HelmCharts, dockerfile etc. Not in this ticket but just generally.

File: eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/EspAuthnRequestAppRuleTest.java [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
 
+    @Before
     public void setup() throws Throwable {
         request = new EidasAuthnRequestBuilder().withIssuer(CONNECTOR_NODE_ENTITY_ID);
     }
 
-    public void shouldReturnRequestIdAndIssuer() throws XPathExpressionException, MarshallingException, TransformerException{
+    @Test
+    @Ignore
+    public void shouldReturnRequestIdAndIssuer() throws Exception {
```
Comment (JHjava) [inline eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/EspAuthnRequestAppRuleTest.java]: There's a ticket to look into the ESP integration tests which I think Dan may have started so should be resolved soon.

File: proxy-node-translator/src/dist/config.yml (hunk 1 above)
Comment (vixus0) [reply proxy-node-translator/src/dist/config.yml]: We'll probably be overriding this whole YAML block in the Helm charts.

File: eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/base/EidasSamlParserAppRuleTestBase.java (hunk 2 above)
Comment (vixus0) [reply eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/base/EidasSamlParserAppRuleTestBase.java]: Yup, taking them out.

File: proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/configuration/TranslatorConfiguration.java (hunk 3 above)
Comment (vixus0) [reply proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/configuration/TranslatorConfiguration.java]: Yeah I've already done some of the cleanup in the Dockerfile and will do the Helm chart stuff in a separate PR.

File: eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/EspAuthnRequestAppRuleTest.java (hunk 4 above)
Comment (vixus0) [reply eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/EspAuthnRequestAppRuleTest.java]: :+1:

**Pull request merged**